finesse/gui/temp_control.py

```diff
@@ -41,138 +34,147 @@ def _create_controls(self) -> QGridLayout:
-        self._ax2.set_ylim([0, 10])
+        self._ax["cold"] = self._ax["hot"].twinx()
+        self._ax["cold"].plot(t, cold_bb_temp, color=cold_colour)
+        self._ax["cold"].set_ylabel("COLD BB", color=cold_colour)
+        self._ax["cold"].set_ylim([0, 10])
 
         self._canvas.draw()
 
-    def _update_figure(self) -> None:  # , x, y) -> None:
-        """Updates the matplotlib figure to be contained within the panel.
+    def _btn_pressed(self, name) -> None:
+        """Shows or hides BB plots."""
+        state = self._ax[name].yaxis.get_visible()
+        self._btns[name].setFlat(state)
+        self._ax[name].yaxis.set_visible(not state)
+        self._ax[name].lines[0].set_visible(not state)
+        self._canvas.draw()
 
-        Args:
-            #x: time
-            #y: temperature
-            # probably won't be inputs since obtainable from elsewhere
-        Returns:
-            None
-        """
-        xdata = list(self._bb_hot_line[0].get_xdata())
-        y1data = list(self._bb_hot_line[0].get_ydata())
-        y2data = list(self._bb_cold_line[0].get_ydata())
+    def _update_figure(self, event) -> None:
+        """Updates the matplotlib figure to be contained within the panel."""
+        xdata = list(self._ax["hot"].lines[0].get_xdata())
+        y1data = list(self._ax["hot"].lines[0].get_ydata())
+        y2data = list(self._ax["cold"].lines[0].get_ydata())
 
         xdata.pop(0)
         y1data.pop(0)
         y2data.pop(0)
 
-        x = xdata[-1] + 10
-
-        # Basic RNG for testing
-        y1 = (60 * y1data[-1] + 50) % 70
-        y2 = (5 * y2data[-1] + 1) % 8
+        if xdata[-1] is not None:
+            x = xdata[-1] + 10
+
+            # Basic RNG for testing
+            y1 = (60 * y1data[-1] + 50) % 70
+            y2 = (5 * y2data[-1] + 1) % 8
+            marker = None
+            linestyle = "-"
+        else:  # adding first data point
+            x = 10
+            y1 = 50
+            y2 = 5
+            marker = "."
+            linestyle = "None"
 
         xdata.append(x)
         y1data.append(y1)
         y2data.append(y2)
 
-        self._bb_hot_line[0].set_xdata(xdata)
-        self._bb_hot_line[0].set_ydata(y1data)
-        self._bb_cold_line[0].set_xdata(xdata)
-        self._bb_cold_line[0].set_ydata(y2data)
-
-        self._ax1.relim()
-        self._ax2.xaxis.axes.relim()
-        self._ax1.autoscale()
-        self._ax2.xaxis.axes.autoscale()
-        self._ax2.set_ylim([self._ax2.get_ylim()[0], 2 * self._ax2.get_ylim()[1]])
+        self._ax["hot"].lines[0].set_xdata(xdata)
+        self._ax["hot"].lines[0].set_ydata(y1data)
+        self._ax["cold"].lines[0].set_xdata(xdata)
+        self._ax["cold"].lines[0].set_ydata(y2data)
+
+        self._ax["cold"].lines[0].set_linestyle(linestyle)
+        self._ax["hot"].lines[0].set_linestyle(linestyle)
+        self._ax["cold"].lines[0].set_marker(marker)
+        self._ax["hot"].lines[0].set_marker(marker)
```

Comment:
Do you need to explicitly set these every time you update the plot? Assuming you only need to do it once, it should be done in the constructor.

Reply:
My thinking was that when there is only one data point (or pair of points), `plt.plot` can't draw a line and so just draws the marker. But yeah, I definitely shouldn't be setting these each time the figure is updated, and it doesn't seem worth doing an `if-else` check every time.
I've now changed this so that the marker and linestyle are set in the constructor. I guess there are both scientific and aesthetic reasons for joining up measured data or not, but that's probably not critical at this stage.

Reply:
Good stuff. If the researchers aren't happy with how it looks we can always tweak it.